internal/certgen/certgen.go

```diff
@@ -75,7 +75,7 @@ func WriteSecretsYAML(outputDir, namespace string, certdata map[string][]byte) e
 
 // WriteSecretsKube writes all the keypairs out to Kube Secrets in the
 // passed Kube context.
-func WriteSecretsKube(client *kubernetes.Clientset, namespace string, certdata map[string][]byte) error {
+func WriteSecretsKube(client kubernetes.Interface, namespace string, certdata map[string][]byte) error {
```

Comment:
This looks like an unrelated change, I think this file and the one above can be pulled out into their own PR to make this simpler to review.

Reply:
See the previous comment about testing with fakes.